dockstore-integration-testing/src/test/java/io/dockstore/client/cli/BasicIT.java

```diff
@@ -1259,35 +1260,316 @@ public void testManualPublishGitlabDocker() {
+        }
+
+        // Manual publish a public Amazon ECR tool that has a repo name with slashes
+        tool = manualRegisterAndPublish(toolsApi, "abcd1234", "foo/bar", null,
+                "git@github.com:DockstoreTestUser/dockstore-whalesay.git", "/Dockstore.cwl", "/Dockstore.wdl", "/Dockerfile",
+                DockstoreTool.RegistryEnum.AMAZON_ECR, "master", "latest", true);
+
+        try {
+            foundTool = toolsApi.getContainerByToolPath(tool.getToolPath(), "");
+            assertEquals(tool.getId(), foundTool.getId());
+        } catch (ApiException e) {
+            fail("Should have been able to get the Amazon ECR tool by tool path.");
+        }
+
+        try {
+            foundTool = toolsApi.getPublishedContainerByToolPath(tool.getToolPath(), "");
+            assertEquals(tool.getId(), foundTool.getId());
+        } catch (ApiException e) {
+            fail("Should have been able to get the published Amazon ECR tool by tool path.");
+        }
+
+        // Manual publish a public Amazon ECR tool that has a repo name with slashes and a tool name
+        tool = manualRegisterAndPublish(toolsApi, "abcd1234", "foo/bar", "test-tool-name",
+                "git@github.com:DockstoreTestUser/dockstore-whalesay.git", "/Dockstore.cwl", "/Dockstore.wdl", "/Dockerfile",
+                DockstoreTool.RegistryEnum.AMAZON_ECR, "master", "latest", true);
+
+        try {
+            foundTool = toolsApi.getContainerByToolPath(tool.getToolPath(), "");
+            assertEquals(tool.getId(), foundTool.getId());
+        } catch (ApiException e) {
+            fail("Should have been able to get the Amazon ECR tool by tool path.");
+        }
+
+        try {
+            foundTool = toolsApi.getPublishedContainerByToolPath(tool.getToolPath(), "");
+            assertEquals(tool.getId(), foundTool.getId());
+        } catch (ApiException e) {
+            fail("Should have been able to get the published Amazon ECR tool by tool path.");
+        }
+
+        try {
+            List<DockstoreTool> foundTools = toolsApi.getContainerByPath(tool.getPath());
+            assertEquals("Should have two tools with the path 'public.ecr.aws/abcd1234/foo/bar'.", 2, foundTools.size());
+        } catch (ApiException e) {
+            fail("Should have been able to get the Amazon ECR tools by path.");
+        }
+
+        try {
+            List<DockstoreTool> foundTools = toolsApi.getPublishedContainerByPath(tool.getPath());
+            assertEquals("Should have two published tools with the path 'public.ecr.aws/abcd1234/foo/bar'.", 2, foundTools.size());
+        } catch (ApiException e) {
+            fail("Should have been able to get the published Amazon ECR tools by path.");
+        }
+    }
+
+    /**
+     * This tests that entry paths are split into their registry, org, repo, and entry name components correctly.
+     */
+    @Test
+    public void testSplitPath() {
```

Comment:
@coverbeck here are the unit tests for `splitPath(String path, boolean hasEntryName)`

Reply:
Whoops, I was looking for them alongside other unit tests, not in the integration tests.